### ISSUE TITLE
Update Application Broker to support Knative events mesh layer

### ISSUE THREAD

**Issue by Abd4llA** [kyma-project/kyma#6148]
**Description**
In order to start introducing the new Knative eventing mesh, Application Broker plays a key part when a non bindable service instance gets provisioned in a namespace by performing the following duties:

1. Enabling namespace eventing via creating the default broker
    (e.g. add the knative-eventing-injection=true label)
2. Wiring the namespace mesh with the Kyma-integration mesh via creating a Knative `Subscription` object from the proper `Event Service` Knative `Channel` to the namespace `Broker` using its URI.

![knative_brokers_co_exist_arch](https://user-images.githubusercontent.com/24314423/67940207-8506d700-fbd3-11e9-8a2f-1249561e5562.png)

**Breakdown**
Left over work starting 11.11.2019
- [X] Unit tests
-  ~~E2e~~ (not in scope)
-  ~~Backup verification~~ (not in scope)
-  ~~Upgradability~~ (not in scope)
-  ~~Documentation~~ (handled here https://github.com/kyma-project/kyma/issues/6312)
- [x] Revert the downgrade Kyma commit
- [x] Change the Knative subscription object api group
- [x] Test

**AC**
1. The application broker behaves as described above.
1. The old behavior is still preserved. 
1. The new behavior have sufficient tests.
    1. ~~E2e (External Solution Test)~~ (not in scope)
    1. ~~Backup~~ (not in scope)
1. ~~Documentation is updated~~ (handled here https://github.com/kyma-project/kyma/issues/6312).

**Comment by bszwarc** [comment]
For the documentation part:
- we need to update the diagrams and steps here: https://kyma-project.io/docs/components/application-connector#architecture-application-broker
- the descriptions for diagrams should also be updated + they should reference the solution description included in eventing documentation (there would be a separate task for that).

**Comment by sayanh** [comment]
@Abd4llA @antoineco @nachtmaar 
With respect to the picture above, is this mapping correct?
ApplicationSourceAdapter = HTTPSource
Broker(default) = Knative-eventing default broker (As per Ralf's comment)

**Comment by sayanh** [comment]
Upgrade:
Alpha scope:
- After upgrade, there is no need to have the new event bus infra for the old subscriptions.

Later scope:
- After upgrade, there needs to be a new event bus infra for the old subscriptions as well.

**Comment by marcobebway** [comment]
This is blocked as it depends on https://github.com/kyma-project/kyma/issues/6259.

**Comment by antoineco** [comment]
@sayanh
ApplicationSource* -> HTTPSource*
The broker is indeed the default Broker injected by Knative.

**Comment by marcobebway** [comment]
The Application Broker documentation changes are done in this pull request: https://github.com/kyma-project/kyma/pull/6454.

**Comment by marcobebway** [comment]
Blocked by: https://github.com/kyma-project/kyma/issues/6259.

**Comment by nachtmaar** [comment]
Unblocked by https://github.com/kyma-project/kyma/pull/6266.
Resource PR is about to be merged: https://github.com/kyma-project/kyma/pulls/montaro
